server/aws-lsp-codewhisperer/src/language-server/codeWhispererSecurityScanServer.ts

```diff
@@ -96,45 +130,83 @@ export const SecurityScanServerToken =
                     { total: 0, withFixes: 0 }
                 )
                 logging.log(`Security scan totally found ${total} issues. ${withFixes} of them have fixes.`)
+                securityScanTelemetryEntry.codewhispererCodeScanTotalIssues = total
+                securityScanTelemetryEntry.codewhispererCodeScanIssuesWithFixes = withFixes
+                scanHandler.throwIfCancelled(token)
 
                 /**
                  * Step 6: send results to diagnostics
                  */
-                // Todo: update diagnostics to add securityRecommendationCollection
+                await diagnosticsProvider.createDiagnostics(securityRecommendationCollection)
 
                 logging.log(`Security scan completed.`)
+                truncation.scannedFiles.forEach(file => logging.log(`Scanned file: ${file}`))
                 return {
                     result: {
                         status: jobStatus,
+                        scannedFiles: Array.from(truncation.scannedFiles.values()).join(','),
                     },
                 } as SecurityScanResponseParams
             } catch (error) {
+                if (error instanceof SecurityScanCancelledError) {
+                    logging.log(`Security scan has been cancelled. ${error}`)
+                    securityScanTelemetryEntry.result = 'Cancelled'
+                    return {
+                        result: {
+                            status: 'Cancelled',
+                        },
+                    } as SecurityScanResponseParams
+                }
                 logging.log(`Security scan failed. ${error}`)
-                // Todo: notify client about scan job failure.
+                securityScanTelemetryEntry.result = 'Failed'
                 return {
                     result: {
                         status: 'Failed',
                     },
                     error,
                 } as SecurityScanResponseParams
+            } finally {
+                securityScanTelemetryEntry.duration = performance.now() - securityScanStartTime
```

Comment:
We won't reach to this part since we are returning in `try/catch` only. Can we pull this up or create return object and return it after this?

Reply:
we will reach finally even if we return from try/catch